### PR TITLE
"Set average weekly working hour" button in user tab made responsive and "Facility registration form cover photo upload" added in create facility

### LINKED ISSUE #7297: Facility registration form- cover photo upload
**Is your feature request related to a problem? Please describe.**
When registering a facility, user gives all details of the facility in the facility registration form (https://care.coronasafe.in/facility/create). However, this form has no option to add facility cover image.

The facility cover image is uploaded after the user submits the facility registration form with the details and is taken to the facility dashboard (https://care.coronasafe.in/facility/31809579-8e0d-402a-b137-843fd77f6290) where the option to upload facility cover image appears.

**Describe the solution you'd like**
Add option to upload facility cover image in the facility registration form itself. This shall be a non-mandatory item.

Retain the option to upload cover image in the facility dashboard as is.

### LINKED ISSUE #7309: "Set average weekly working hour" button in user tab is not responsive
**Describe the bug**

"Set average weekly working hour" button text is clipping out in user tab ( Mobile View) 

**To Reproduce**

1. Go to https://care.coronasafe.in/users
2. switch to responsive view and see the error

**Expected behavior**

it should be properly responsive in mobile view

**Screenshots**

image

### PR DESCRIPTION
**Pull request by arnit-coder**
## Proposed Changes

- Fixes #7309 
- Fixes #7297 
- "Set average weekly working hour" in user tab is now responsive.
- "Upload cover photo field" added in facility registration form but the backend part is not added. Need to update the backend part also.

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] image
This image is before i corrected the issue, in this image the button "set average weekly working hour is not responsive"
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Correction of Issue #7309 
image
The button is responsive now.
-[ ] Correction of Issue #7297 
image
The upload cover photo field
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Review by Ashesh3 — APPROVED** [review]
LGTM, but fix the lint errors:

`npm run lint-fix`

File: src/Components/Facility/FacilityCreate.tsx [hunk 1]
```diff
@@ -158,7 +168,55 @@ export const FacilityCreate = (props: FacilityProps) => {
...
+  const hasCoverImage = !!facilityData?.read_cover_image_url;
+
+  const StaffUserTypeIndex = USER_TYPES.findIndex((type) => type === "Staff");
+  const hasPermissionToEditCoverImage =
+    !(authUser.user_type as string).includes("ReadOnly") &&
+    USER_TYPES.findIndex((type) => type == authUser.user_type) >=
+    StaffUserTypeIndex;
+  // const hasPermissionToDeleteFacility =
+  //   authUser.user_type === "DistrictAdmin" ||
+  //   authUser.user_type === "StateAdmin";
```
Comment (Ashesh3) [inline src/Components/Facility/FacilityCreate.tsx:200]: Remove the commented code if not needed

File: src/Components/Facility/FacilityCreate.tsx [hunk 2]
```diff
@@ -827,6 +885,63 @@ export const FacilityCreate = (props: FacilityProps) => {
...
+                        >
+                          <path d="M18.5 6C18.5 5.4475 18.95 5 19.5 5H20.5C21.05 5 21.5 5.4475 21.5 6V7.5H23C23.55 7.5 24 7.95 24 8.5V9.5C24 10.05 23.55 10.5 23 10.5H21.5V12C21.5 12.55 21.05 13 20.5 13H19.5C18.95 13 18.5 12.55 18.5 12V10.5H17C16.45 10.5 16 10.05 16 9.5V8.5C16 7.95 16.45 7.5 17 7.5H18.5V6ZM25.5 0C27.9875 0 30 2.015 30 4.5V5H35.5C37.9875 5 40 7.0125 40 9.5V27.5C40 29.9875 37.9875 32 35.5 32H4.49875C2.01188 32 0 29.9875 0 27.5V9.5C0 7.0125 2.015 5 4.5 5H10V4.5C10 2.015 12.0125 0 14.5 0H25.5ZM30 8V29H35.5C36.3312 29 37 28.3313 37 27.5V21H33.5C32.6688 21 32 20.3313 32 19.5C32 18.6688 32.6688 18 33.5 18H37V15H33.5C32.6688 15 32 14.3313 32 13.5C32 12.6688 32.6688 12 33.5 12H37V9.5C37 8.66875 36.3312 8 35.5 8H30ZM3 9.5V12H6.5C7.33125 12 8 12.6688 8 13.5C8 14.3313 7.33125 15 6.5 15H3V18H6.5C7.33125 18 8 18.6688 8 19.5C8 20.3313 7.33125 21 6.5 21H3V27.5C3 28.3313 3.67125 29 4.49875 29H10V8H4.5C3.67188 8 3 8.66875 3 9.5ZM13 29H17V25C17 23.3438 18.3438 22 20 22C21.6562 22 23 23.3438 23 25V29H27V4.5C27 3.67188 26.3312 3 25.5 3H14.5C13.6688 3 13 3.67188 13 4.5V29Z" />
+                        </svg>
```
Comment (Ashesh3) [inline src/Components/Facility/FacilityCreate.tsx:940]: Use CareIcon component for this icon

File: package.json [hunk 3]
```diff
@@ -71,6 +72,7 @@
     "echarts-for-react": "^3.0.2",
     "eslint-mdx": "^3.1.5",
     "events": "^3.3.0",
+    "googlemaps": "^1.12.0",
```
Comment (rithviknishad) [inline package.json:75]: Why was this package added?

File: src/Components/Facility/FacilityCreate.tsx [hunk 4]
```diff
@@ -135,16 +144,17 @@ const facilityCreateReducer = (state = initialState, action: FormAction) => {
...
   const { t } = useTranslation();
   const { gov_data_api_key, kasp_string, kasp_enabled } = useConfig();
-  const { facilityId } = props;
-
   const [state, dispatch] = useAutoSaveReducer<FacilityForm>(
     facilityCreateReducer,
     initialState
   );
-  const [isLoading, setIsLoading] = useState(false);
+  const [isLoading1, setIsLoading1] = useState(false);
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx:157]: Give meaningful variable names. One cannot infer what `isLoading1` is used for.

File: src/Components/Facility/FacilityCreate.tsx [hunk 5]
```diff
@@ -158,7 +168,55 @@ export const FacilityCreate = (props: FacilityProps) => {
...
+  useMessageListener((data) => console.log(data));
+
+  const {
+    data: facilityData,
+    loading: isLoading,
+    refetch: facilityFetch,
+  } = useQuery(routes.getPermittedFacility, {
+    pathParams: {
+      id: facilityId || "",
+    },
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx:183]: set prefetch to false if no facility id is present.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Facility/FacilityCreate.tsx (hunk 5 above)
Comment (arnit-coder) [reply src/Components/Facility/FacilityCreate.tsx:183]: Respected sir,
I have my mid term exams going on.
As soon as I will get time I will work on the requested changes.

File: src/Components/Facility/FacilityCreate.tsx (hunk 2 above)
Comment (arnit-coder) [reply src/Components/Facility/FacilityCreate.tsx:940]: Sir, while inserting the CareIcon component i found that a dot is coming when this is inserted

File: src/Components/Facility/FacilityCreate.tsx (hunk 1 above)
Comment (arnit-coder) [reply src/Components/Facility/FacilityCreate.tsx:200]: Removed sir.

File: package.json (hunk 3 above)
Comment (arnit-coder) [reply package.json:75]: Removed.

**Pull request closed without merging**